src/pages-and-resources/discussions/app-config-form/apps/shared/discussion-restrictions/CollapseCardHeading.jsx

```diff
@@ -0,0 +1,44 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { Badge } from '@edx/paragon';
+
+const CollapseCardHeading = ({
+  isOpen,
+  expandHeadingText,
+  collapseHeadingText,
+  badgeVariant,
+  badgeStatus,
+}) => {
+  const ExpandHeadingTextComp = useMemo(() => {
+    if (isOpen) {
+      return <span className="h4 py-2 mr-auto">{expandHeadingText}</span>;
+    }
+    return null;
+  }, [isOpen, expandHeadingText]);
+
+  if (ExpandHeadingTextComp) {
+    return ExpandHeadingTextComp;
+  }
```

Comment:
It is just a single html. we can avoid memorization here. Over memorization also kills performance. Only apply memorization to complex calculation and looping operations.